tools/src/main/java/com/nedap/archie/archetypevalidator/ArchetypeValidator.java

```diff
@@ -104,26 +105,27 @@ public ValidationResult validate(Archetype archetype, FullArchetypeRepository re
         if(settings == null) {
             settings = new ArchetypeValidationSettings();
         }
+        OverridingInMemFullArchetypeRepository extraRepository = null;
+        if(repository == null) {
+            extraRepository = new OverridingInMemFullArchetypeRepository();
+        } else {
+            extraRepository = new OverridingInMemFullArchetypeRepository(repository);
+        }
+        repository = extraRepository;
         if(archetype instanceof Template) {
             //in the case of a template, add a repository that can store the overlays separate from the rest of the archetypes
             //later they can be retrieved and handled as extra archetypes, that are not top level archetypes usable in other
             //archetypes that are not templates
             //TODO: can we specialize templates? In that case we need more work to get any template overlays defined in
             //the parent template in this repo as well
```

Comment:
The comment above needs to be reworded a bit now the `extraRepository` has been moved up.